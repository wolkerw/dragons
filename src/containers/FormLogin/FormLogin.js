import React, { Component } from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
// import { TextLabel } from "../../components/Labels";
import Input from "../../components/Input/Input";
import InputPassword from "../../components/InputPassword/InputPassword";
// import messages from "../../assets/i18n";
import ButtonStandard from "../../components/ButtonStandard/ButtonStandard";
import "./FormLogin.scss";
import "../../styles/base/_typography.scss";

class FormLogin extends Component {
    constructor() {
        super();
        this.state = {
            validInputs: false,
            disabled: true,
            login: "",
            password: "",
            loginIsValid: false,
            passwordIsValid: false,
            firstInteraction: true,
            completedAuth: false
        };

        this.handleSubmit = this.handleSubmit.bind(this);
    }

    saveLoginData(token /*, profiles, chapa*/) {
        localStorage.setItem("TOKEN", token);
        // localStorage.setItem("profiles", JSON.stringify(profiles));
        // localStorage.setItem("chapa", chapa);

        return true;
    }

    componentWillMount() {
        // if is logged, redirect to the main
        try {
            const isLogged = this.props.isLogged();

            if (isLogged && typeof isLogged === "string" && isLogged)
                this.props.history.push(`/main`);
        } catch (e) {
            console.log("FormLogin componentWillMount catch error");
        }
    }

    componentDidUpdate = () => {
        if (this.state.completedAuth) return;

        let gotToken =
            this.props.authentication && this.props.authentication.length;
        /* let gotPermissions =
            this.props.authorization && this.props.authorization.length; */

        // test if the redux updated the props
        if (gotToken) {
            //&& !gotPermissions
            // this.props.getAuthorization(
            //     localStorage.getItem("TOKEN")
            //     // JSON.parse(localStorage.getItem("profiles"))
            // );

            this.setState({ completedAuth: true });
            this.props.history.push(`/main`);
        }
    };

    login = () => {
        if (this.state.completedAuth) return;

        /*let gotToken =
            this.props.authentication && this.props.authentication.length;
        let gotPermissions =
            this.props.authorization && this.props.authorization.length;*/

        this.props.getAuthentication();
    };

    callback = (text, stateName, valid) => {
        this.setState({ [stateName]: text, [stateName + "IsValid"]: valid });
    };

    validationServices = () => {
        //Chamada do serviço de validação
        //simulação do serviço de validação
        //let logged = false;
        //let profiles = [];

        // TODO get token and profiles from authentication saga
        /*if (this.state.login === "admin" && this.state.password === "admin") {
            profiles = ["grp-meta-dev-root", "grp-meta-dev-root"];
        } else if (
            this.state.login === "123456" &&
            this.state.password === "123456"
        ) {
            profiles = ["grp-meta-dev-po"];
        } else if (
            this.state.login === "carga" &&
            this.state.password === "carga"
        ) {
            profiles = ["grp-meta-dev-carga"];
        } else if (
            this.state.login === "conciliacao" &&
            this.state.password === "conciliacao"
        ) {
            profiles = ["mdl-meta-dev-conciliacao"];
        } else if (
            this.state.login === "relatorio" &&
            this.state.password === "relatorio"
        ) {
            profiles = ["mdl-meta-dev-relatorio"];
        } else if (
            this.state.login === "agendamentos" &&
            this.state.password === "agendamentos"
        ) {
            profiles = ["mdl-meta-dev-agendamento"];
        }*/
        //if (profiles.length) logged = true;

        /*if (!logged) {
            return this.setState({
                validInputs: false,
                firstInteraction: false
            });
        }*/

        this.setState({ validInputs: true, firstInteraction: true });

        const token =
            "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJsb2dpbiI6Im1ldGEiLCJzZW5oYSI6Im1ldGEifQ.dBDm1QTX-Nfu-eciOCh76z6LXzRLl-8hnmwm11zzcZc";

        // TODO a chapa deve vir do back
        this.saveLoginData(token /*, profiles this.state.login "888001"*/);

        this.login();
    };

    handleSubmit(event) {
        event.preventDefault();
        this.validationServices();
    }

    render() {
        const { lang, loginIsRequired } = this.props;
        const {
            firstInteraction,
            validInputs,
            loginIsValid,
            passwordIsValid
        } = this.state;

        return (
            <form onSubmit={this.handleSubmit}>
                <div type="text" className="form-group">
                    <Input
                        placeholder="Login"
                        validateCallback={this.callback}
                        firstSubmit={firstInteraction}
                        isRequired={loginIsRequired}
                        name="login"
                    />
                </div>
                <div className="form-group">
                    <InputPassword
                        placeholder="Senha"
                        validateCallback={this.callback}
                        firstSubmit={firstInteraction}
                    />
                </div>
                <div className="error-message text-1">
                    {!validInputs && !firstInteraction ? (
                        // <TextLabel id={messages[lang.name].loginError} />
                        <p>Dados incorretos!</p>
                    ) : (
                        ""
                    )}
                </div>
                <div
                    className={
                        validInputs && !firstInteraction
                            ? "error-button"
                            : "form-group-button"
                    }
                >
                    <ButtonStandard
                        disabled={
                            !(loginIsRequired
                                ? loginIsValid && passwordIsValid
                                : passwordIsValid)
                        }
                        textId="Entrar"
                        className={
                            "btn-login btn-rounded mt-2" +
                            (!(loginIsRequired
                                ? loginIsValid && passwordIsValid
                                : passwordIsValid)
                                ? " btn-login-disabled"
                                : " btn-hover")
                        }
                        type="submit"
                    />
                </div>
            </form>
        );
    }
}

const mapStateToProps = state => {
    return {
        authentication: state.authentication,
        //authorization: state.authorization,
        lang: state.language
    };
};

const mapDispatchToProps = dispatch => {
    return {
        getAuthentication: () =>
            dispatch({
                type: "GET_AUTHENTICATION"
            }) /*,
        getAuthorization: (token, profiles) =>
            dispatch({
                type: "GET_AUTHORIZATION",
                token: token,
                profiles: profiles
            })*/
    };
};

// export for testing and storybook
export const FormLoginWithoutWithRouter = connect(
    mapStateToProps,
    mapDispatchToProps
)(FormLogin);

// export for use
export default withRouter(
    connect(
        mapStateToProps,
        mapDispatchToProps
    )(FormLogin)
);
