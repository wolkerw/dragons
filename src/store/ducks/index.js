import { combineReducers } from "redux";

import login from "./login";
import input from "./login";
import dragons from "./dragons";
import dragonList from "./dragonList";

export default combineReducers({
    login,
    dragons,
    dragonList,
    input
});
