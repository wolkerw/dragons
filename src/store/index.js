import { createStore, applyMiddleware } from "redux";
import createSagaMiddleware from "redux-saga";
import { composeWithDevTools } from 'redux-devtools-extension'
import reducers from './ducks';
import rootSaga from "./sagas";

/**
 * Indice dos Ducks e Sagas
 * @author FÃ¡bio Sartori
 * @copyright 06/2019
 */
const sagaMiddleware = createSagaMiddleware();

let midCompose = "";

if (process.env.NODE_ENV === "development") {
    midCompose = composeWithDevTools(
        applyMiddleware(sagaMiddleware)
        );
    } else {
    midCompose = applyMiddleware(sagaMiddleware);
}

const store = createStore(reducers, midCompose);

sagaMiddleware.run(rootSaga);

export default store;