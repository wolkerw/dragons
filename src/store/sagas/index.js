import { all } from "redux-saga/effects";
// import { getFreeGatesWatcher } from "./freeGates";
import { getOrdersWatcher } from "./orders";
import { getAuthenticationWatcher } from "./authentication";
// import { getAuthorizationWatcher } from "./authorization";
// import { getReconcileWatcher } from "./reconcile";
//import { getReport } from "./relatorio";

export default function* rootSaga() {
    yield all([
        // getFreeGatesWatcher(),
        getOrdersWatcher(),
        getAuthenticationWatcher()
        // getAuthorizationWatcher(),
        // getReconcileWatcher()
        //getReport()
    ]);
}
