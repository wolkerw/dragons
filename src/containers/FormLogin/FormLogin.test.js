import React from "react";
import { FormLoginWithoutWithRouter } from "./FormLogin";
import { Provider } from "react-redux";
import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import ptTranslations from "../../assets/i18n/messages-pt.json";
import { IntlProvider } from "react-intl";
import store from "./../../store";

configure({ adapter: new Adapter() });
const country = "en-US";

const isLogged = () => {
    return localStorage.getItem("TOKEN");
};

describe("FormLogin Component", () => {
    const component = (
        <IntlProvider
            locale={country}
            key={country}
            messages={ptTranslations}
            defaultLocale={country}
        >
            <Provider store={store}>
                <FormLoginWithoutWithRouter isLogged={isLogged} />
            </Provider>
        </IntlProvider>
    );

    it("matches the snapshot", () => {
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("generated a form with two inputs and a button inside", () => {
        const wrapper = mount(component);

        const input = wrapper.find("FormLogin Input input").length;
        expect(input).toEqual(1);

        const inputPassword = wrapper.find("FormLogin InputPassword input")
            .length;
        expect(inputPassword).toEqual(1);

        const loginButton = wrapper.find("FormLogin ButtonStandard button")
            .length;
        expect(loginButton).toEqual(1);
    });
});
