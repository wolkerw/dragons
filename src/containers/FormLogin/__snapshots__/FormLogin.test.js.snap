// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLogin Component matches the snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
    type="text"
  >
    <div
      className="input"
      onBlur={[Function]}
    >
      <input
        autoComplete="off"
        className="form-control form-control"
        name="login"
        onChange={[Function]}
        placeholder="Login"
        type="text"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <div
      className="input-password"
      onBlur={[Function]}
      type="text"
    >
      <input
        autoComplete="off"
        className="form-control form-control"
        name="password"
        onChange={[Function]}
        placeholder="Senha"
        type="password"
      />
    </div>
  </div>
  <div
    className="error-message text-1"
  >
    
  </div>
  <div
    className="form-group-button"
  >
    <button
      className="btn-login btn-rounded mt-2 btn-login-disabled"
      disabled={true}
      type="submit"
    >
      <span>
        Login
      </span>
      
    </button>
  </div>
</form>
`;
